src/Host/Client/Impl/rtvs/R/util.R

```diff
@@ -198,3 +198,25 @@ rmarkdown_publish <- function(blob_id, output_format, encoding) {
     rmarkdown::render(rmdpath, output_format = output_format, output_file = output_filepath,  output_dir = tempdir(), encoding = encoding);
     readBin(output_filepath, 'raw', file.info(output_filepath)$size);
 }
+
+package_lock_state <- function(package_name, lib_path) {
+    files <- dir(path = paste0(lib_path,'/', package_name), pattern = '\\.', recursive = TRUE, ignore.case = TRUE, full.names = TRUE)
+    lock_state <- call_embedded('get_file_lock_state', files);
+    factor(lock_state, levels = c(0, 1, 2), labels = c('unlocked', 'locked_by_r_session', 'locked_by_other'));
```

Comment:
Can you make a helper "definition" for this outside of the function, so that it's easy to locate and reuse? Something like this:
```r
as.lock_state <- function(x)
  factor(x, levels = c(0, 1, 2), labels = c('unlocked', 'locked_by_r_session', 'locked_by_other'));
```
and then it can be used like the standard conversion functions:
```r
as.lock_state(state)
```
And this can be our standard pattern for enums in R.